etl/steps/data/garden/eth/2023-03-15/ethnic_power_relations.py

```diff
@@ -0,0 +1,247 @@
+        "onset_do_gov_noneth_flag",
+    ]
+
+    df = df[["country", "year"] + epr_vars + acd2epr_vars]
+
+    # Multiply share variables by 100
+    share_vars = [
+        "lpop",
+        "egippop",
+        "legippop",
+        "exclpop",
+        "lexclpop",
+        "discrimpop",
+        "ldiscrimpop",
+        "maxexclpop",
+        "lmaxexclpop",
+        "regautpop",
+        "regautexclpop",
+        "regautegippop",
+    ]
+
+    df[share_vars] *= 100
+
+    return df
+
+
+def load_population() -> pd.DataFrame:
+    # Load population table from key_indicators dataset.
+    ds_indicators: Dataset = paths.load_dependency("population")
+    tb_population = ds_indicators["population"]
+    tb_population = pd.DataFrame(tb_population).reset_index(drop=False)
+
+    return tb_population
+
+
+# Function to add regional aggregations to data
+def add_regional_aggregations(df: pd.DataFrame) -> pd.DataFrame:
+
+    # Load population data
+    tb_population = load_population()
+
+    # Merge df with tb_population to get population
+    df = pd.merge(df, tb_population[["country", "year", "population"]], how="left", on=["country", "year"])
+
+    # Define variables which their aggregation is weighted by population
+    # Do not aggregate: legippop, lexclpop, ldiscrimpop, lmaxexclpop
+    pop_vars = ["lpop", "egippop", "exclpop", "discrimpop", "maxexclpop", "regautpop", "regautexclpop", "regautegippop"]
+
+    # Multiply variables by population
+    pop_vars_headcount = []
+    for var in pop_vars:
+        pop_vars_headcount.append(f"{var}_headcount")
+        df[f"{var}_headcount"] = df[var] * df["population"] / 100
+
+    # Define regions to aggregate
+    regions = [
+        "Europe",
+        "Asia",
+        "North America",
+        "South America",
+        "Africa",
+        "Oceania",
+        "High-income countries",
+        "Low-income countries",
+        "Lower-middle-income countries",
+        "Upper-middle-income countries",
+        "European Union (27)",
+        "World",
+    ]
+
+    # Define the variables and aggregation method to be used in the following function loop
+    aggregations = {
+        "egip_groups_count": "sum",
+        "excl_groups_count": "sum",
+        "regaut_groups_count": "sum",
+        "regaut_excl_groups_count": "sum",
+        "regaut_egip_groups_count": "sum",
+        "rlvt_groups_count": "sum",
+        "actv_groups_count": "sum",
+        "lpop_headcount": "sum",
+        "egippop_headcount": "sum",
+        "exclpop_headcount": "sum",
+        "discrimpop_headcount": "sum",
+        "maxexclpop_headcount": "sum",
+        "regautpop_headcount": "sum",
+        "regautexclpop_headcount": "sum",
+        "regautegippop_headcount": "sum",
+        "population": "sum",
+    }
+
+    # Add regional aggregates, by summing up the variables in `aggregations`
+    for region in regions:
+        df = geo.add_region_aggregates(df, region=region, aggregations=aggregations, population=df)
```

Comment:
Just so you know, this function is a bit "too smart" and does a few things you may not be aware of (which is why we need to create a better one). If the most populated countries in a region are not informed, the aggregate will be nan. If you don't want this behaviour, add the argument `countries_that_must_have_data=[]`.

Reply:
I saw that in the documentation and it doesn't generate much problem because there is data for almost every country in the world.

Reply:
And regions are currently fixed in time, right? For instance, if for the 80s I don't have Russia or the other former Soviet republics will it not calculate the aggregation?